pkg/metrics/metrics.go

```diff
@@ -44,6 +45,10 @@ const (
 // CreateMetricsService creates a Kubernetes Service to expose the passed metrics
 // port(s) with the given name(s).
 func CreateMetricsService(ctx context.Context, cfg *rest.Config, servicePorts []v1.ServicePort) (*v1.Service, error) {
+	if _, err := k8sutil.GetOperatorNamespace(); err != nil && errors.Is(err, k8sutil.ErrRunLocal) {
+		log.Info("Skipping create service metric services; not running in a cluster.")
+		return nil, nil
+	}
```

Comment:
I don't agree with this change. I don't think we should put special handling for `up local` in a library package. Ditto for the changes to `CreateServiceMonitors`.

Instead can we limit changes to handle this in the `main.go` scaffold only?

Reply:
Done @joelanford .